compiler.ts

```diff
@@ -190,7 +208,42 @@ function codeGenExpr(expr: Expr<Type>, env: GlobalEnv): Array<string> {
 function codeGenValue(val: Value<Type>, env: GlobalEnv): Array<string> {
   switch (val.tag) {
     case "num":
-      return ["(i32.const " + val.value + ")"];
+      var x = BigInt(val.value) // for division
+      var n = 0
+      var digits : Number[] = []
+      while(x != BigInt(0)) {
+          if (x < 0) {
+            x *= BigInt(-1)
+          }
+          digits.push(Number(x & BigInt(0x7fffffff)))
+          x = x / BigInt(1 << 31) 
+          n = n + 1
+      }
+      n = n + 1 // store (n+1) blocks (n: number of digits)
+
+      var i = 0
+      var return_val : string[] = []
+      
+      return_val.push(`(i32.const ${n})`);
+      return_val.push(`(call $alloc)`);
+      return_val.push(`(local.set $$scratch)`);
+      
+      // store the bignum in (n+1) blocks
+      for (i; i < n; i++) {
+        if (i == 0) {
```

Comment:
This is awkward. Why put this special case inside the loop? Could just start `i` at 1 for the loop, for example. Also, if you want it in the loop to share the get/const/store lines, those shouldn't be repeated.